sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -428,53 +432,85 @@ class SessionCatalog(
    */
   def dropFunction(name: FunctionIdentifier): Unit = {
     val db = name.database.getOrElse(currentDb)
+    val qualified = name.copy(database = Some(db)).unquotedString
+    if (functionRegistry.functionExists(qualified)) {
+      // If we have loaded this function into FunctionRegistry,
+      // also drop it from there.
+      functionRegistry.dropFunction(qualified)
```

Comment:
I don't understand why we have to do this. Isn't the function registry only for temporary functions? We only put things in it in `createTempFunction`, so we just only remove things from it in `dropTempFunction`. This method here is for persistent functions only.